jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannelOverHttp.java

```diff
@@ -500,31 +501,24 @@ public void parsedHeader(HttpField field)
 
                 case EXPECT:
                 {
-                    if (HttpVersion.HTTP_1_1.equals(_requestBuilder.version()))
+                    if (!Index.parseCsvIndex(HttpHeaderValue.CACHE, value, t ->
```

Comment:
The `parseCsvIndex` method looks very specific to the `HttpChannelOverHttp`. It feels like it should be moved into this class.

Reply:
Moved and fixed multiple bugs.
There are some other possible usages of the method, but will do those in another PR